File: data/EXP01/python/2594/2594.output_1.py
# @lc app=leetcode id=2594 lang=python3
from typing import List

class Solution:
    def repairCars(self, ranks: List[int], cars: int) -> int:
        ranks.sort()  # sort ranks in ascending order
        total_time = 0  # initialize total time to 0
        
        for rank in reversed(ranks):
            num_cars = rank * rank  # calculate number of cars the mechanic can repair
            
            if num_cars >= cars:
                total_time += cars * rank * rank  # repair the remaining cars
                break  # exit the loop
                
            total_time += num_cars * rank * rank  # repair num_cars cars
            cars -= num_cars  # decrease the number of cars remaining
        
        return total_time

File: data/EXP01/golang_no_system_var/2594/2594.output_1.go
func repairCars(ranks []int, cars int) int64 {
    sort.Sort(sort.Reverse(sort.IntSlice(ranks)))

    totalTime := int64(0)

    carsRepaired := make([]int, len(ranks))

    for i := 0; i < cars; i++ {
        minTime := int64(1<<63 - 1)
        mechanicIdx := -1
        for j, rank := range ranks {
            time := (int64(rank) * int64(carsRepaired[j]+1) * int64(carsRepaired[j]+1))
            if time < minTime {
                minTime = time
                mechanicIdx = j
            }
        }

        carsRepaired[mechanicIdx]++
        totalTime = max(totalTime, minTime)
    }

    return totalTime
}

func max(a, b int64) int64 {
    if a > b {
        return a
    }
    return b
}